IntelliTect.Multitool/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Text;
+
+namespace IntelliTect.Multitool.Extensions;
+
+/// <summary>
+/// Various string extensions
+/// </summary>
+public static class StringExtensions
+{
+    /// <summary>
+    /// Slugify's text so that it is URL compatible. IE: "Hi My Name is" -> "hi-my-name-is".
+    /// Removes special characters, sets to lowercase, replaces ' ' and '_' with '-', and trims the string
+    /// </summary>
+    public static string Slugify(this string str)
+    {
+        str = str.ToLowerInvariant().Trim();
+        StringBuilder sb = new();
+        const char separatorCharacter = '-';
+        bool allowSeparator = false;
+        foreach (char character in str)
+        {
+            switch (character)
+            {
+                // this second '-' here is different than a normal - in terms of key code
+                // so we replace it with a normal -
+                case char c when c == '_' || c == ' ' || c == '–' || c == '-' || c == '.' || c == ',':
+                    if (allowSeparator)
+                    {
+                        sb.Append(separatorCharacter);
+                        allowSeparator = false;
+                    }
+                    break;
+                // Only allow letters and numbers as valid characters (removing things like diacritics (accents))
+                case char c when (c >= '0' && c <= '9') || (c >= 'a' && c <= 'z'):
+                    sb.Append(character);
+                    allowSeparator = true;
+                    break;
+                default:
+                    break;
+            }
+        }
+        return sb.ToString().TrimEnd(separatorCharacter);
```

Comment:
You can avoid this trim call by switching the algorithm to be a for loop so you know if you are at the end or not.